src/shared/reboot-util.c

```diff
@@ -12,9 +12,12 @@
 #include "umask-util.h"
 #include "virt.h"
 
-int update_reboot_parameter_and_warn(const char *parameter) {
+int update_reboot_parameter_and_warn(const char *parameter, bool keep) {
         int r;
 
+        if (keep && !parameter)
+                return 0;
```

Comment:
why not move this into the next if block, it checks `parameter` too after all, and hence no need to check it again (it checks it slightly different though, with isempty() rather than != NULL, but this should be more accurate anyway)

Reply:
Ok. Changed as suggested. Works similarly with that way too.